Regression: DESCRIBE / SHOW COLUMNS fails on PostgreSQL tables with Geometry columns

### Trino version

481

### Please describe the bug

Starting in Trino version `481`, executing a `DESCRIBE` statement or querying `information_schema.columns` on a PostgreSQL table containing a PostGIS geometry column fails with an `INTERNAL_ERROR` (`GENERIC_INTERNAL_ERROR`). This appears to be a regression introduced by the migration from the ESRI geometry library to JTS in PR #27881.

Steps to Reproduce
Create a table in a PostgreSQL database with a PostGIS geometry column:
```
CREATE TABLE public.areas (
    id SERIAL PRIMARY KEY,
    geom geometry(Geometry, 4326)
);
```

Configure a PostgreSQL catalog (e.g., `postgres`).

Attempt to inspect the table schema from the Trino CLI:

```
DESCRIBE postgres.public.areas;
-- Or run: SHOW COLUMNS FROM postgres.public.areas;
```

Expected Behavior
Trino should successfully describe the table schema and map the PostGIS column to the native Trino `Geometry` type.

Actual Behavior
The query fails instantly with the following error:

```
Error Type:  INTERNAL_ERROR
Error Code: GENERIC_INTERNAL_ERROR (65536)

com.google.common.util.concurrent.UncheckedExecutionException: com.google.common.util.concurrent.UncheckedExecutionException: java.lang.IllegalArgumentException: Trino type Geometry is not compatible with read function io.trino.plugin.jdbc.ObjectReadFunction$1@510c5995 returning class org.locationtech.jts.geom.Geometry
	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2077)
	at com.google.common.cache.LocalCache.get(LocalCache.java:3985)
	at com.google.common.cache.LocalCache$LocalManualCache.get(LocalCache.java:4854)
	at io.trino.cache.EvictableCache.get(EvictableCache.java:120)
	at io.trino.plugin.jdbc.CachingJdbcClient.get(CachingJdbcClient.java:840)
	at io.trino.plugin.jdbc.CachingJdbcClient.getColumns(CachingJdbcClient.java:185)
	at io.trino.plugin.jdbc.DefaultJdbcMetadata.getTableMetadata(DefaultJdbcMetadata.java:1082)
	at io.trino.plugin.base.classloader.ClassLoaderSafeConnectorMetadata.getTableMetadata(ClassLoaderSafeConnectorMetadata.java:283)
	at io.trino.tracing.TracingConnectorMetadata.getTableMetadata(TracingConnectorMetadata.java:253)
	at io.trino.metadata.MetadataManager.getTableMetadata(MetadataManager.java:531)
	at io.trino.tracing.TracingMetadata.getTableMetadata(TracingMetadata.java:332)
	at io.trino.sql.rewrite.ShowQueriesRewrite$Visitor.visitShowColumns(ShowQueriesRewrite.java:492)
	at io.trino.sql.rewrite.ShowQueriesRewrite$Visitor.visitShowColumns(ShowQueriesRewrite.java:226)
	at io.trino.sql.tree.ShowColumns.accept(ShowColumns.java:58)
	at io.trino.sql.tree.AstVisitor.process(AstVisitor.java:27)
	at io.trino.sql.rewrite.ShowQueriesRewrite.rewrite(ShowQueriesRewrite.java:223)
	at io.trino.sql.rewrite.StatementRewrite.rewrite(StatementRewrite.java:54)
	at io.trino.sql.analyzer.Analyzer.analyze(Analyzer.java:93)
	at io.trino.sql.analyzer.Analyzer.analyze(Analyzer.java:87)
	at io.trino.execution.SqlQueryExecution.analyze(SqlQueryExecution.java:288)
	at io.trino.execution.SqlQueryExecution.<init>(SqlQueryExecution.java:222)
	at io.trino.execution.SqlQueryExecution$SqlQueryExecutionFactory.createQueryExecution(SqlQueryExecution.java:893)
	at io.trino.dispatcher.LocalDispatchQueryFactory.lambda$createDispatchQuery$0(LocalDispatchQueryFactory.java:163)
	at io.trino.$gen.Trino_481____20260517_060906_2.call(Unknown Source)
	at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:128)
	at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:74)
	at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:80)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
	at java.base/java.lang.Thread.run(Thread.java:1474)
Caused by: com.google.common.util.concurrent.UncheckedExecutionException: java.lang.IllegalArgumentException: Trino type Geometry is not compatible with read function io.trino.plugin.jdbc.ObjectReadFunction$1@510c5995 returning class org.locationtech.jts.geom.Geometry
	at io.trino.cache.EmptyCache.get(EmptyCache.java:101)
	at io.trino.plugin.jdbc.CachingJdbcClient.get(CachingJdbcClient.java:840)
	at io.trino.plugin.jdbc.CachingJdbcClient.getColumns(CachingJdbcClient.java:185)
	at io.trino.plugin.jdbc.CachingJdbcClient.lambda$getColumns$0(CachingJdbcClient.java:185)
	at com.google.common.cache.LocalCache$LocalManualCache$1.load(LocalCache.java:4859)
	at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3556)
	at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2307)
	at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2180)
	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2071)
	... 29 more
Caused by: java.lang.IllegalArgumentException: Trino type Geometry is not compatible with read function io.trino.plugin.jdbc.ObjectReadFunction$1@510c5995 returning class org.locationtech.jts.geom.Geometry
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:484)
	at io.trino.plugin.jdbc.ColumnMapping.<init>(ColumnMapping.java:114)
	at io.trino.plugin.jdbc.ColumnMapping.objectMapping(ColumnMapping.java:92)
	at io.trino.plugin.postgresql.PostgreSqlClient.geometryColumnMapping(PostgreSqlClient.java:1912)
	at io.trino.plugin.postgresql.PostgreSqlClient.toColumnMapping(PostgreSqlClient.java:594)
	at io.trino.plugin.postgresql.PostgreSqlClient.getColumns(PostgreSqlClient.java:499)
	at io.trino.plugin.jdbc.ForwardingJdbcClient.getColumns(ForwardingJdbcClient.java:112)
	at io.trino.plugin.jdbc.jmx.StatisticsAwareJdbcClient.lambda$getColumns$0(StatisticsAwareJdbcClient.java:134)
	at io.trino.plugin.jdbc.jmx.JdbcApiStats.wrap(JdbcApiStats.java:34)
	at io.trino.plugin.jdbc.jmx.StatisticsAwareJdbcClient.getColumns(StatisticsAwareJdbcClient.java:134)
	at io.trino.plugin.jdbc.RetryingJdbcClient.lambda$getColumns$0(RetryingJdbcClient.java:109)
	at dev.failsafe.Functions.lambda$toCtxSupplier$11(Functions.java:243)
	at dev.failsafe.Functions.lambda$get$0(Functions.java:46)
	at dev.failsafe.internal.RetryPolicyExecutor.lambda$apply$0(RetryPolicyExecutor.java:74)
	at dev.failsafe.SyncExecutionImpl.executeSync(SyncExecutionImpl.java:187)
	at dev.failsafe.FailsafeExecutor.call(FailsafeExecutor.java:376)
	at dev.failsafe.FailsafeExecutor.get(FailsafeExecutor.java:112)
	at io.trino.plugin.jdbc.RetryingModule.retry(RetryingModule.java:67)
	at io.trino.plugin.jdbc.RetryingJdbcClient.getColumns(RetryingJdbcClient.java:109)
	at io.trino.plugin.jdbc.CachingJdbcClient.lambda$getColumns$0(CachingJdbcClient.java:185)
	at io.trino.cache.EmptyCache.get(EmptyCache.java:95)
	... 37 more
```

--- a/plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/ColumnMapping.java
+++ b/plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/ColumnMapping.java
@@ -113,14 +113,16 @@ public final class ColumnMapping
         this.writeFunction = requireNonNull(writeFunction, "writeFunction is null");
         checkArgument(
                 type.getJavaType() == readFunction.getJavaType(),
-                "Trino type %s is not compatible with read function %s returning %s",
+                "Trino type %s with Java type %s is not compatible with read function %s returning %s",
                 type,
+                type.getJavaType(),
                 readFunction,
                 readFunction.getJavaType());
         checkArgument(
                 type.getJavaType() == writeFunction.getJavaType(),
-                "Trino type %s is not compatible with write function %s accepting %s",
+                "Trino type %s with Java type %s is not compatible with write function %s accepting %s",
                 type,
+                type.getJavaType(),
                 writeFunction,
                 writeFunction.getJavaType());
         this.predicatePushdownController = requireNonNull(predicatePushdownController, "predicatePushdownController is null");

--- a/plugin/trino-postgresql/pom.xml
+++ b/plugin/trino-postgresql/pom.xml
@@ -43,11 +43,6 @@
 
         <dependency>
             <groupId>io.trino</groupId>
-            <artifactId>trino-geospatial-toolkit</artifactId>
-        </dependency>
-
-        <dependency>
-            <groupId>io.trino</groupId>
             <artifactId>trino-matching</artifactId>
         </dependency>
 
@@ -69,11 +64,6 @@
         <dependency>
             <groupId>org.jdbi</groupId>
             <artifactId>jdbi3-core</artifactId>
-        </dependency>
-
-        <dependency>
-            <groupId>org.locationtech.jts</groupId>
-            <artifactId>jts-core</artifactId>
         </dependency>
 
         <dependency>

--- a/plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java
+++ b/plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java
@@ -117,7 +117,6 @@ import io.trino.spi.type.TypeSignature;
 import io.trino.spi.type.VarcharType;
 import org.jdbi.v3.core.Handle;
 import org.jdbi.v3.core.Jdbi;
-import org.locationtech.jts.geom.Geometry;
 import org.postgresql.core.TypeInfo;
 import org.postgresql.jdbc.PgConnection;
 
@@ -159,8 +158,6 @@ import static com.google.common.collect.ImmutableList.toImmutableList;
 import static com.google.common.collect.ImmutableMap.toImmutableMap;
 import static io.airlift.slice.Slices.utf8Slice;
 import static io.airlift.slice.Slices.wrappedBuffer;
-import static io.trino.geospatial.serde.JtsGeometrySerde.deserialize;
-import static io.trino.geospatial.serde.JtsGeometrySerde.serialize;
 import static io.trino.plugin.base.util.JsonTypeUtil.jsonParse;
 import static io.trino.plugin.base.util.JsonTypeUtil.toJsonValue;
 import static io.trino.plugin.jdbc.DecimalSessionSessionProperties.getDecimalDefaultScale;
@@ -1911,39 +1908,52 @@ public class PostgreSqlClient
     {
         return ColumnMapping.objectMapping(
                 geometryType,
-                ObjectReadFunction.of(Geometry.class, (resultSet, columnIndex) -> {
-                    String hexWkb = resultSet.getString(columnIndex);
-                    byte[] wkb = HexFormat.of().parseHex(hexWkb);
-                    return deserialize(wrappedBuffer(wkb));
-                }),
-                geometryWriteFunction(),
+                new ObjectReadFunction()
+                {
+                    @Override
+                    public Class<?> getJavaType()
+                    {
+                        return geometryType.getJavaType();
+                    }
+
+                    @Override
+                    public Object readObject(ResultSet resultSet, int columnIndex)
+                            throws SQLException
+                    {
+                        String hexWkb = resultSet.getString(columnIndex);
+                        byte[] wkb = HexFormat.of().parseHex(hexWkb);
+                        Slice slice = wrappedBuffer(wkb);
+
+                        BlockBuilder blockBuilder = geometryType.createBlockBuilder(null, 1);
+                        geometryType.writeSlice(blockBuilder, slice);
+                        return geometryType.getObject(blockBuilder.build(), 0);
+                    }
+                },
+                new ObjectWriteFunction()
+                {
+                    @Override
+                    public Class<?> getJavaType()
+                    {
+                        return geometryType.getJavaType();
+                    }
+
+                    @Override
+                    public String getBindExpression()
+                    {
+                        return "ST_GeomFromEWKB(?)";
+                    }
+
+                    @Override
+                    public void set(PreparedStatement statement, int index, Object value)
+                            throws SQLException
+                    {
+                        BlockBuilder blockBuilder = geometryType.createBlockBuilder(null, 1);
+                        geometryType.writeObject(blockBuilder, value);
+                        Slice slice = geometryType.getSlice(blockBuilder.build(), 0);
+                        statement.setBytes(index, slice.getBytes());
+                    }
+                },
                 DISABLE_PUSHDOWN);
-    }
-
-    private static ObjectWriteFunction geometryWriteFunction()
-    {
-        return new ObjectWriteFunction()
-        {
-            @Override
-            public Class<?> getJavaType()
-            {
-                return Geometry.class;
-            }
-
-            @Override
-            public String getBindExpression()
-            {
-                return "ST_GeomFromEWKB(?)";
-            }
-
-            @Override
-            public void set(PreparedStatement statement, int index, Object value)
-                    throws SQLException
-            {
-                Geometry geometry = (Geometry) value;
-                statement.setBytes(index, serialize(geometry).getBytes());
-            }
-        };
     }
 
     private static class StatisticsDao

--- a/testing/trino-product-tests-groups/src/main/java/io/trino/tests/product/TestGroups.java
+++ b/testing/trino-product-tests-groups/src/main/java/io/trino/tests/product/TestGroups.java
@@ -65,6 +65,7 @@ public final class TestGroups
     public static final String EXASOL = "exasol";
     public static final String CASSANDRA = "cassandra";
     public static final String POSTGRESQL = "postgresql";
+    public static final String POSTGRESQL_POSTGIS = "postgresql_postgis";
     public static final String SQLSERVER = "sqlserver";
     public static final String LDAP = "ldap";
     public static final String LDAP_AND_FILE = "ldap_and_file";

--- a/testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvMultinodePostgresqlPostgis.java
+++ b/testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvMultinodePostgresqlPostgis.java
@@ -1,0 +1,75 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.trino.tests.product.launcher.env.environment;
+
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Inject;
+import io.trino.tests.product.launcher.docker.DockerFiles;
+import io.trino.tests.product.launcher.env.DockerContainer;
+import io.trino.tests.product.launcher.env.Environment;
+import io.trino.tests.product.launcher.env.EnvironmentProvider;
+import io.trino.tests.product.launcher.env.common.StandardMultinode;
+import io.trino.tests.product.launcher.env.common.TestsEnvironment;
+import io.trino.tests.product.launcher.testcontainers.PortBinder;
+import org.testcontainers.containers.startupcheck.IsRunningStartupCheckStrategy;
+
+import static io.trino.tests.product.launcher.docker.ContainerUtil.forSelectedPorts;
+import static java.util.Objects.requireNonNull;
+import static org.testcontainers.utility.MountableFile.forHostPath;
+
+@TestsEnvironment
+public final class EnvMultinodePostgresqlPostgis
+        extends EnvironmentProvider
+{
+    // Use non-default PostgreSQL port to avoid conflicts with locally installed PostgreSQL if any.
+    public static final int POSTGRESQL_PORT = 15432;
+
+    private final DockerFiles dockerFiles;
+    private final PortBinder portBinder;
+
+    @Inject
+    public EnvMultinodePostgresqlPostgis(StandardMultinode standardMultinode, DockerFiles dockerFiles, PortBinder portBinder)
+    {
+        super(ImmutableList.of(standardMultinode));
+        this.dockerFiles = requireNonNull(dockerFiles, "dockerFiles is null");
+        this.portBinder = requireNonNull(portBinder, "portBinder is null");
+    }
+
+    @Override
+    public void extendEnvironment(Environment.Builder builder)
+    {
+        builder.addConnector(
+                "postgresql",
+                forHostPath(dockerFiles.getDockerFilesHostPath("conf/environment/multinode-postgresql/postgresql.properties")));
+        builder.addContainer(createPostgreSql());
+    }
+
+    @SuppressWarnings("resource")
+    private DockerContainer createPostgreSql()
+    {
+        // Use the oldest supported PostgreSQL version
+        DockerContainer container = new DockerContainer("postgis/postgis:11-3.2", "postgresql")
+                .withEnv("POSTGRES_PASSWORD", "test")
+                .withEnv("POSTGRES_USER", "test")
+                .withEnv("POSTGRES_DB", "test")
+                .withEnv("PGPORT", Integer.toString(POSTGRESQL_PORT))
+                .withStartupCheckStrategy(new IsRunningStartupCheckStrategy())
+                .waitingFor(forSelectedPorts(POSTGRESQL_PORT));
+
+        portBinder.exposePort(container, POSTGRESQL_PORT);
+
+        return container;
+    }
+}

--- a/testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/suite/suites/Suite7NonGeneric.java
+++ b/testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/suite/suites/Suite7NonGeneric.java
@@ -18,6 +18,7 @@ import io.trino.tests.product.launcher.env.EnvironmentConfig;
 import io.trino.tests.product.launcher.env.EnvironmentDefaults;
 import io.trino.tests.product.launcher.env.environment.EnvMultinodeMinioDataLakeTaskRetriesFilesystem;
 import io.trino.tests.product.launcher.env.environment.EnvMultinodePostgresql;
+import io.trino.tests.product.launcher.env.environment.EnvMultinodePostgresqlPostgis;
 import io.trino.tests.product.launcher.env.environment.EnvMultinodePostgresqlSpooling;
 import io.trino.tests.product.launcher.env.environment.EnvMultinodeSecretsProvider;
 import io.trino.tests.product.launcher.env.environment.EnvMultinodeSpooling;
@@ -42,6 +43,7 @@ import static io.trino.tests.product.TestGroups.HIVE_SPARK;
 import static io.trino.tests.product.TestGroups.HIVE_SPARK_NO_STATS_FALLBACK;
 import static io.trino.tests.product.TestGroups.JOIN;
 import static io.trino.tests.product.TestGroups.POSTGRESQL;
+import static io.trino.tests.product.TestGroups.POSTGRESQL_POSTGIS;
 import static io.trino.tests.product.TestGroups.SMOKE;
 import static io.trino.tests.product.TestGroups.SQLSERVER;
 import static io.trino.tests.product.TestGroups.STORAGE_FORMATS;
@@ -64,6 +66,9 @@ public class Suite7NonGeneric
                         .build(),
                 testOnEnvironment(EnvMultinodePostgresqlSpooling.class)
                         .withGroups(CONFIGURED_FEATURES, POSTGRESQL)
+                        .build(),
+                testOnEnvironment(EnvMultinodePostgresqlPostgis.class)
+                        .withGroups(CONFIGURED_FEATURES, POSTGRESQL_POSTGIS)
                         .build(),
                 testOnEnvironment(EnvMultinodeSecretsProvider.class)
                         .withGroups(CONFIGURED_FEATURES, POSTGRESQL)

--- a/testing/trino-product-tests/src/main/java/io/trino/tests/product/postgresql/TestPostgresqlGeometryType.java
+++ b/testing/trino-product-tests/src/main/java/io/trino/tests/product/postgresql/TestPostgresqlGeometryType.java
@@ -1,0 +1,74 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.tests.product.postgresql;
+
+import io.trino.tempto.AfterMethodWithContext;
+import io.trino.tempto.BeforeMethodWithContext;
+import io.trino.tempto.ProductTest;
+import org.testng.annotations.Test;
+
+import static io.trino.tempto.assertions.QueryAssert.Row.row;
+import static io.trino.tests.product.TestGroups.POSTGRESQL_POSTGIS;
+import static io.trino.tests.product.TestGroups.PROFILE_SPECIFIC_TESTS;
+import static io.trino.tests.product.utils.QueryExecutors.onPostgres;
+import static io.trino.tests.product.utils.QueryExecutors.onTrino;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestPostgresqlGeometryType
+        extends ProductTest
+{
+    private static final String TABLE_NAME = "public.areas";
+
+    @BeforeMethodWithContext
+    @AfterMethodWithContext
+    public void dropTestTable()
+    {
+        onPostgres().executeQuery("DROP TABLE IF EXISTS " + TABLE_NAME);
+    }
+
+    @Test(groups = {POSTGRESQL_POSTGIS, PROFILE_SPECIFIC_TESTS})
+    public void testReadGeometryFromPostgres()
+    {
+        onPostgres().executeQuery("CREATE TABLE " + TABLE_NAME + " (id SERIAL PRIMARY KEY, geom geometry(Geometry, 4326))");
+        onPostgres().executeQuery("INSERT INTO " + TABLE_NAME + " (geom) VALUES (ST_SetSRID(ST_Point(1, 2), 4326))");
+
+        assertThat(onTrino().executeQuery("SELECT ST_AsText(geom) FROM postgresql." + TABLE_NAME))
+                .containsOnly(row("POINT (1 2)"));
+    }
+
+    @Test(groups = {POSTGRESQL_POSTGIS, PROFILE_SPECIFIC_TESTS})
+    public void testWriteGeometryFromTrino()
+    {
+        onPostgres().executeQuery("CREATE TABLE " + TABLE_NAME + " (id SERIAL PRIMARY KEY, geom geometry(Geometry, 4326))");
+
+        onTrino().executeQuery("INSERT INTO postgresql." + TABLE_NAME + " (geom) VALUES (ST_GeometryFromText('POINT (3 4)'))");
+
+        assertThat(onPostgres().executeQuery("SELECT ST_AsText(geom) FROM " + TABLE_NAME))
+                .containsOnly(row("POINT(3 4)"));
+        assertThat(onTrino().executeQuery("SELECT ST_AsText(geom) FROM postgresql." + TABLE_NAME))
+                .containsOnly(row("POINT (3 4)"));
+    }
+
+    @Test(groups = {POSTGRESQL_POSTGIS, PROFILE_SPECIFIC_TESTS})
+    public void testNullGeometry()
+    {
+        onPostgres().executeQuery("CREATE TABLE " + TABLE_NAME + " (id SERIAL PRIMARY KEY, geom geometry(Geometry, 4326))");
+        onPostgres().executeQuery("INSERT INTO " + TABLE_NAME + " (geom) VALUES (NULL), (ST_SetSRID(ST_Point(5, 6), 4326))");
+
+        assertThat(onTrino().executeQuery("SELECT id FROM postgresql." + TABLE_NAME + " WHERE geom IS NULL"))
+                .containsOnly(row(1));
+        assertThat(onTrino().executeQuery("SELECT ST_AsText(geom) FROM postgresql." + TABLE_NAME + " WHERE geom IS NOT NULL"))
+                .containsOnly(row("POINT (5 6)"));
+    }
+}

--- a/testing/trino-product-tests/src/main/java/io/trino/tests/product/utils/QueryExecutors.java
+++ b/testing/trino-product-tests/src/main/java/io/trino/tests/product/utils/QueryExecutors.java
@@ -90,6 +90,11 @@ public final class QueryExecutors
         return testContext().getDependency(QueryExecutor.class, "mysql");
     }
 
+    public static QueryExecutor onPostgres()
+    {
+        return testContext().getDependency(QueryExecutor.class, "postgres");
+    }
+
     public static QueryExecutor onSpark()
     {
         return new QueryExecutor()

